src/core/expression/qgsexpression.cpp

```diff
@@ -885,11 +885,11 @@ void QgsExpression::initVariableHelp()
 
 
   //symbol variables
-  sVariableHelpTexts()->insert( QStringLiteral( "geometry_part_count" ), QCoreApplication::translate( "variable_help", "Number of parts in rendered feature's geometry." ) );
-  sVariableHelpTexts()->insert( QStringLiteral( "geometry_part_num" ), QCoreApplication::translate( "variable_help", "Current geometry part number for feature being rendered." ) );
-  sVariableHelpTexts()->insert( QStringLiteral( "geometry_ring_num" ), QCoreApplication::translate( "variable_help", "Current geometry ring number for feature being rendered (for polygon features only). The exterior ring has a value of 0." ) );
-  sVariableHelpTexts()->insert( QStringLiteral( "geometry_point_count" ), QCoreApplication::translate( "variable_help", "Number of points in the rendered geometry's part. It is only meaningful for line geometries and for symbol layers that set this variable." ) );
-  sVariableHelpTexts()->insert( QStringLiteral( "geometry_point_num" ), QCoreApplication::translate( "variable_help", "Current point number in the rendered geometry's part. It is only meaningful for line geometries and for symbol layers that set this variable." ) );
+  sVariableHelpTexts()->insert( QStringLiteral( "geometry_part_count" ), QCoreApplication::translate( "variable_help", "Number of parts in rendered feature's geometry. To force rendering of entire feature, you can disable 'Clip features to canvas extent' under Advanced symbol settings." ) );
```

Comment:
Thanks @phidrho for lookinh into these details.
If I understand correctly, it does not return by default "all the parts in the visible feature", but instead "all the actually visible parts in that feature".
Would a formulation along this line be clearer, i.e. moving the visibility over the item and keeping focus on the items "counting" and not feature rendering.
```suggestion
  sVariableHelpTexts()->insert( QStringLiteral( "geometry_part_count" ), QCoreApplication::translate( "variable_help", "Number of rendered parts in feature's geometry. To get all parts of the geometry, disable 'Clip features to canvas extent' under Advanced symbol settings." ) );
```
Or replace "rendered" with "visible" ? And/Or advise using a function (e.g. num_geometries or whatever function is suitable)?
_PS: I didn't try to apply reformulation to the other variables, yet._